projects/element-ng/typeahead/si-typeahead.search.ts

```diff
@@ -0,0 +1,274 @@
+/**
+ * Copyright (c) Siemens 2016 - 2025
+ * SPDX-License-Identifier: MIT
+ */
+import { computed, Signal } from '@angular/core';
+
+export interface SearchOptions<T> {
+  /** The text that will be used for searching. */
```

Comment:
Specially here. I first thought this is the text, that the user enters. But no. The is the test that is used to match the search. It is constructed on the value of the option. I guess it could event be translated of a date object be transformed to a string.